dojo/vulnerability/__init__.py

```diff
@@ -0,0 +1,97 @@
+import json
+import logging
+import re
+from pathlib import Path
+from typing import Iterator, Optional, Iterable
+
+import git
+from django.conf import settings
+from django.db import transaction
+
+from dojo.models import Vulnerability
+
+
+class VulnerabilityMirror:
+    """Manages a git repository mirror of vulnerability data."""
+
+    __slots__ = ('_remote_url', '_repository_root', '_repo', '_logger')
+
+    def __init__(self,
+                 remote_url: str,
+                 repository_root: str = settings.VULNDB_ROOT,
+                 logger: logging.Logger = logging.getLogger(__name__)):
+        self._remote_url = remote_url
+        self._repository_root = Path(repository_root)
+        self._logger = logger
+        if not self._repository_root.exists():
+            self._repository_root.mkdir(parents=True)
+        git_dir = self._repository_root / '.git'
+        if not git_dir.exists():
+            self._logger.debug('Cloning vulnerability mirror from %s to %s', remote_url, repository_root)
+            self._repo = self.__clone()
+        else:
+            self._logger.debug('Reusing existing checkout at %s', repository_root)
+            self._repo = self.__update()
+
+    def __clone(self) -> git.Repo:
+        return git.Repo.clone_from(self._remote_url, self._repository_root, depth=1)
```

Comment:
This significantly speeds up the initial clone, but now this can introduce a potential issue where the last processed commit is older than the clone on the worker.